_posts/2020-08-20-flink-docker.md

```diff
@@ -0,0 +1,90 @@
+
+  The new images support passing configuration variables via a `FLINK_PROPERTIES` environment variable. Users can enable default plugins with the `ENABLE_BUILT_IN_PLUGINS`environment variable. The images also allow loading custom jar paths and configuration files.
+
+Looking into the future, there are already some interesting potential improvements lined up: 
+
+- [Java 11 Docker images](https://issues.apache.org/jira/browse/FLINK-16260) (already completed)
+- [Use vanilla docker-entrypoint with flink-kubernetes](https://issues.apache.org/jira/browse/FLINK-15793) (in progress)
+- [History server support](https://issues.apache.org/jira/browse/FLINK-17167)
+- [Support for OpenShift](https://issues.apache.org/jira/browse/FLINK-15587)
+
+## How do I get started?
+
+This is a short tutorial on how to start a Flink Session Cluster with docker.
+
+A *Flink Session cluster* can be used to run multiple jobs. Each job needs to be submitted to the cluster after it has been deployed. To deploy a *Flink Session cluster* with Docker, you need to start a *JobManager* container. To enable communication between the containers, we first set a required Flink configuration property and create a network:
+
+```
+FLINK_PROPERTIES="jobmanager.rpc.address: jobmanager"
+docker network create flink-network
+```
+
+Then we launch the JobManager:
+
+```
+docker run \
+       --rm \
+       --name=jobmanager \
+       --network flink-network \
+       -p 8081:8081 \
+       --env FLINK_PROPERTIES="${FLINK_PROPERTIES}" \
+       flink:1.11.1 jobmanager
+```
+and one or more *TaskManager* containers:
+
+```
+docker run \
+      --rm \
+      --name=taskmanager \
+      --network flink-network \
+      --env FLINK_PROPERTIES="${FLINK_PROPERTIES}" \
+      flink:1.11.1 taskmanager
+```
+
+You now have a fully functional Flink cluster running! You can access the the web front end here: [localhost:8081](http://localhost:8081/).
+
+Let's now submit one of Flink's example jobs:
+
+```bash
+# 1: (optional) Download the Flink distribution, and unpack it
+wget https://archive.apache.org/dist/flink/flink-1.11.1/flink-1.11.1-bin-scala_2.12.tgz
+tar xf flink-1.11.1-bin-scala_2.12.tgz
+cd flink-1.11.1
+
+# 2: Start the Flink job
+./bin/flink run ./examples/streaming/TopSpeedWindowing.jar
+```
+
+<center>
+<img src="{{ site.baseurl }}/img/blog/flink-docker/flink-docker.gif" width="882px" height="730px" alt="Demo video"/>
+</center>
+
```

Comment:
I would also briefly mention that one could also use docs link to deploy the job mode, just to denote that it is also possible and we envisioned it.